cpp/src/arrow/compute/api_aggregate.cc

```diff
@@ -41,5 +41,9 @@ Result<Datum> MinMax(const Datum& value, const MinMaxOptions& options, ExecConte
   return CallFunction("min_max", {value}, &options, ctx);
 }
 
+Result<Datum> Mode(const Datum& value, ExecContext* ctx) {
+  return CallFunction("mode", {value}, ctx);
```

Comment:
We have a comprehensive doc for compute functions here:
https://arrow.apache.org/docs/cpp/compute.html

You should add your mode function to `docs/source/cpp/compute.rst`.

Reply:
Done